### PR TITLE
Various refactorings, prepping to work on timeline overhaul

### PR DESCRIPTION
**Pull request by Graywing13**
NOTE: I've temporarily put the preferences stuff in the sidebar (bc I need to use preferencesfor the timeline overhaul). But as we discussed, this will eventually be a modal. I just didn't want to make a modal while Meng was working on it. 

- Add toggle
- Add basic preferences 
- Change map background when in trip/day view
- Refactor buttons to not have default color, to avoid colouring issues
- Remove default exports

### CODE REVIEW

File: client/src/components/user/Logout.js [hunk 1]
```diff
@@ -10,13 +11,11 @@ export function Logout() {
   }
 
   return (
-    <div>
-      <Button
-        className='fixed bottom-2 bg-red-500 bg-opacity-80 text-white hover:bg-red-300'
```
Comment (NLmeng) [inline client/src/components/user/Logout.js:15]: looks good. i dunno about window, but on my laptop i think this hover is too light

**Review by NLmeng — APPROVED** [review]

File: client/src/components/sideBar/SideBar.js [hunk 2]
```diff
@@ -36,31 +45,63 @@ export function SideBar() {
...
+          label='Vertical Timelines'
+          onClick={() => {
+            dispatch(toggleVerticalTimelines())
+          }}
+          active={isVerticalTimelines}
+        />
+      </div>
+    )
+  }, [isVerticalTimelines, isCompactView])
+
```
Comment (NLmeng) [inline client/src/components/sideBar/SideBar.js:92]: also for this one, do you mean (in the future) that we will need to place the toggles inside the modal?

**Review by vee-16 — APPROVED** [review]
lgtm

**Review by vee-16 — APPROVED** [review]
lgtm

File: client/src/components/sideBar/SideBar.js (hunk 2 above)
Comment (NLmeng) [reply client/src/components/sideBar/SideBar.js:92]: nvm i forgot to read your comment lmao

**Pull request merged**